src/turbostat.c

```diff
@@ -944,7 +941,10 @@ static int __attribute__((warn_unused_result)) probe_cpu(void) {
   /*
    * Enable or disable C states depending on the model and family
    */
-  if (family == 6) {
+  if (vendor_id[0] == 0x756e6547 /* "Genu" */ &&
+      vendor_id[1] == 0x49656e69 /* "ineI" */ &&
+      vendor_id[2] == 0x6c65746e /* "ntel" */ &&
+      family == 6) {
```

Comment:
Maybe use a union with char array where byte[12] is zeroed, and a string comparison?